### PR TITLE
interface: item availability

### PR DESCRIPTION
**Pull request by BadrAly**
* Adapts existing item availability.
* Provides availability reasons when item is not available.

Co-Authored-by: Laurent Dubois <laurent.dubois@uclouvain.be>
Co-Authored-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: rero_ils/modules/documents/views.py [hunk 1]
```diff
@@ -319,7 +319,8 @@ def item_status_text(item, format='medium', locale='en'):
         text = _('not available')
         if item.status == ItemStatus.ON_LOAN:
             due_date = format_date_filter(
-                item.get_item_end_date(), format=format, locale=locale
+                item.get_item_end_date(
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/documents/views.py]: Done

File: rero_ils/modules/items/api.py [hunk 2]
```diff
@@ -899,3 +899,31 @@ def organisation_view(self):
...
+
+    @property
+    def not_available_reasons(self):
+        """Returns reasons why item is not available."""
+        text = ''
+        if not self.available:
+            if self.status == ItemStatus.ON_LOAN:
+                due_date = format_date_filter(self.get_item_end_date(),
+                                              format='short_date', locale='en')
+                text = '{0} {1}'.format(_('due until'), due_date)
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/items/api.py]: Done

File: rero_ils/modules/items/api_views.py [hunk 3]
```diff
@@ -319,3 +319,16 @@ def item(item_barcode):
...
+@api_blueprint.route('/availabilty/<item_pid>', methods=['GET'])
+@check_authentication
+@jsonify_error
+def item_availability(item_pid):
+    """HTTP GET request for item availability."""
+    item = Item.get_record_by_pid(item_pid)
+    if not item:
+        abort(404)
+    return jsonify({
+        'availability': item.available
```
[the remark this answers is not in the document]
Comment (lauren-d) [reply rero_ils/modules/items/api_views.py]: Not necessary.

File: Pipfile [hunk 4]
```diff
@@ -63,6 +62,7 @@ pytest-runner = ">=3.0.0,<5"
 # FIXME: Remove this once this issue is resolved https://github.com/pypa/pipenv/issues/3865.
 docutils = "==0.15"
 transifex-client = ">=0.12.5"
+pylint = "*"
```
Comment (rerowep) [inline Pipfile]: Dont include `pylintit` it is not needed by `rero-ils` and maybe someone else is using `flake`

File: rero_ils/modules/items/api.py [hunk 5]
```diff
@@ -899,3 +899,14 @@ def organisation_view(self):
         """Get Organisation view for item."""
         organisation = Organisation.get_record_by_pid(self.organisation_pid)
         return organisation['view_code']
+
+    def item_has_active_loan_or_request(self):
+        """Return True if active loan or a request found for item."""
+        states = ['PENDING'] + \
```
Comment (rerowep) [inline rero_ils/modules/items/api.py]: Sometimes better to write with `format`

Comment (lauren-d) [reply rero_ils/modules/items/api.py]: In this part of code, we concatenate two arrays.
Using "format" is not possible.
Maybe I'm wrong.

File: Pipfile (hunk 4 above)
Comment (lauren-d) [reply Pipfile]: Done

**Pull request merged**